Scripts/cours.js

```diff
@@ -1,31 +1,42 @@
-var classes = [
+const rawTab = [
     {
         "id": 1,
         "label": "Algorithmes",
         "period":"01/01 - 01/03/2020",
-        "nbHour": 35+" h",
+        "nbHour": { 
+            "valeur":35,
+            "unity": "heures"
+            },
         "teacher": "Nom Prénom",
         "detail": "Morem ipsum dolor sit amet, consectetur adipisicing elit. Amet numquam aspernatur!"    },
     {
         "id":2,
         "label": "Conception",
         "period": "01/02 - 29/04/2020",
-        "nbHour": 20+" h",
+        "nbHour": { 
+            "valeur":20,
+            "unity": "heures"
+            },
         "teacher": "Nom Prénom",
         "detail": "Morem ipsum dolor sit amet, consectetur adipisicing elit. Amet numquam aspernatur!"
     },
     {
         "id":3,
         "label": "Base de données",
         "period": "01/02 - 15/03/2020",
-        "nbHour": 40+" h",
+        "nbHour": { 
+            "valeur":40,
+            "unity": "heures"
+            },
         "teacher": "Nom Prénom",
         "detail": "Morem ipsum dolor sit amet, consectetur adipisicing elit. Amet numquam aspernatur!"
     }
 ]
 
+var classes  = rawTab;
```

Comment:
by doing this, you juste create another name to the same array, it meanse whenever you update classes the rawTab is updated also. So you have to create a copy of the rawTab and set the classes with this copy. (see slice function)

Reply:
I don't forget this now !!